src/mongocxx/lib/mongocxx/v1/read_concern.cpp

```diff
@@ -12,4 +12,159 @@
+#include <bsoncxx/v1/stdx/string_view.hpp>
+
+#include <cstring>
+#include <string>
+#include <unordered_map>
+
+#include <bsoncxx/private/bson.hh>
+
+#include <mongocxx/private/mongoc.hh>
+#include <mongocxx/private/scoped_bson.hh>
+#include <mongocxx/private/utility.hh>
+
+namespace mongocxx {
+namespace v1 {
+
+namespace {
+
+mongoc_read_concern_t* to_mongoc(void* ptr) {
+    return static_cast<mongoc_read_concern_t*>(ptr);
+}
+
+} // namespace
+
+read_concern::~read_concern() {
+    libmongoc::read_concern_destroy(to_mongoc(_impl));
+}
+
+read_concern::read_concern(read_concern&& other) noexcept : _impl{exchange(other._impl, nullptr)} {}
+
+read_concern& read_concern::operator=(read_concern&& other) noexcept {
+    if (this != &other) {
+        libmongoc::read_concern_destroy(to_mongoc(exchange(_impl, exchange(other._impl, nullptr))));
+    }
+    return *this;
+}
+
+read_concern::read_concern(read_concern const& other) : _impl{libmongoc::read_concern_copy(to_mongoc(other._impl))} {}
+
+read_concern& read_concern::operator=(read_concern const& other) {
+    if (this != &other) {
+        libmongoc::read_concern_destroy(
+            to_mongoc(exchange(_impl, libmongoc::read_concern_copy(to_mongoc(other._impl)))));
+    }
+    return *this;
+}
+
+read_concern::read_concern() : _impl{libmongoc::read_concern_new()} {}
+
+read_concern& read_concern::acknowledge_level(level v) {
+    // Return value is always true: no input validation by mongoc.
+    switch (v) {
+        case level::k_local:
+            (void)libmongoc::read_concern_set_level(to_mongoc(_impl), MONGOC_READ_CONCERN_LEVEL_LOCAL);
+            break;
+        case level::k_majority:
+            (void)libmongoc::read_concern_set_level(to_mongoc(_impl), MONGOC_READ_CONCERN_LEVEL_MAJORITY);
+            break;
+        case level::k_linearizable:
+            (void)libmongoc::read_concern_set_level(to_mongoc(_impl), MONGOC_READ_CONCERN_LEVEL_LINEARIZABLE);
+            break;
+
+        case level::k_server_default:
+            (void)libmongoc::read_concern_set_level(to_mongoc(_impl), nullptr);
+            break;
+
+        case level::k_available:
+            (void)libmongoc::read_concern_set_level(to_mongoc(_impl), MONGOC_READ_CONCERN_LEVEL_AVAILABLE);
+            break;
+        case level::k_snapshot:
+            (void)libmongoc::read_concern_set_level(to_mongoc(_impl), MONGOC_READ_CONCERN_LEVEL_SNAPSHOT);
+            break;
+
+        default:
+        case level::k_unknown:
+            // Precondition violation: undocumented but well-defined behavior.
+            (void)libmongoc::read_concern_set_level(to_mongoc(_impl), nullptr);
+            break;
+    }
+
+    return *this;
+}
+
+read_concern::level read_concern::acknowledge_level() const {
+    auto const level_cstr = libmongoc::read_concern_get_level(to_mongoc(_impl));
+
+    if (!level_cstr) {
+        return level::k_server_default;
+    }
+
+    std::unordered_map<bsoncxx::v1::stdx::string_view, level> map = {
```

Comment:
```suggestion
    static std::unordered_map<bsoncxx::v1::stdx::string_view, level> const map = {
```

Since this is just a lookup table, I believe it can be `static const`.